<?xml version="1.0"?>
<!-- Provided by sleutene for convenience -->
<launch>
  <!-- Launch simulator. -->
  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find sjtu_drone)/launch/simple.launch">
    <arg name="world_name" value="$(find sjtu_drone)/worlds/skokloster_castle.world"/>
    <!--arg name="world_name" value="$(find sjtu_drone)/worlds/garching_kitchen.world"/-->
  </include>
  
  <!-- Emulate controller parameters-->
  <group ns="ardrone_driver">
    <param name="euler_angle_max" value="0.21" />
    <param name="control_vz_max" value="700" />
    <param name="control_yaw" value="1.75" />
  </group>
  
  <!-- Launch visualisation. -->
  <!--node pkg="rviz" type="rviz" name="rviz_vi_ekf" args="-d $(find ardrone_practicals)/config/rviz_vi_ekf.rviz"/-->
  <node pkg="rviz" type="rviz" name="rviz_control" args="-d $(find ardrone_practicals)/config/rviz_control.rviz"/>
  <!-- Launch estimator / control arp_node. -->
  <node name="arp_node" pkg="ardrone_practicals" type="arp_node"
      output="screen">
    <remap from="ardrone/front/image_raw" to="drone/front_camera/image_raw"/>
    <remap from="ardrone/imu" to="drone/imu"/> 
    <remap from="ardrone/takeoff" to="drone/takeoff"/> 
    <remap from="ardrone/reset" to="drone/reset"/>
    <remap from="ardrone/land" to="drone/land"/> 
    <remap from="ardrone/navdata" to="drone/navdata"/> 
    <!-- Camera parameters-->
    <param name="fu" value="185.6909" />
    <param name="fv" value="185.6909" />
    <param name="cu" value="319.5" />
    <param name="cv" value="179." />
    <param name="k1" value="0" />
    <param name="k2" value="0" />
    <param name="p1" value="0" />
    <param name="p2" value="0" />
    <rosparam param="T_SC">[0.0, 0.0, 1.0, 0.2, -1.0, 0.0, 0.0, 0.0, 0.0, -1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0]</rosparam>
    <!-- Make sure to match the map to the world loaded above! -->

    <param name="map" value="okvis2-slam-final_map_skokloster_new2.csv" />
    <!--param name="map" value="okvis2-slam-final_map_garching_new2.csv" /-->
    <param name="occupancymap" value="occupancy-map_skokloster.dat" />
    <!--param name="occupancymap" value="occupancy-map_garching.dat" /-->

    <!-- Parameter for final task -->

    <!-- Starting waypoint -->
    <rosparam param="startPos">[0.0, 0.0, 0.5]</rosparam>
    <!-- Ending waypoint -->
    <!-- Kitchen easy waypoint -->
    <!--rosparam param="goalPos">[0.8 ,-0.7, 1.67]</rosparam-->
    <!-- sko bottom left waypoint -->
    <rosparam param="goalPos">[-2.4 ,11.6, 1.7]</rosparam>
    <!-- sko top left waypoint -->
    <!--rosparam param="goalPos">[5.4 ,10.5, 0.7]</rosparam-->
    <!-- sko middle up waypoint -->
    <!--rosparam param="goalPos">[5.1 ,6.2, 0.7]</rosparam-->
  </node>

</launch>
